test/upgrade/prober/wathola/sender/services.go

```diff
@@ -104,15 +116,52 @@ func NewCloudEvent(data interface{}, typ string) cloudevents.Event {
 	return e
 }
 
+// ResetEventSenders will reset configured event senders to defaults.
+func ResetEventSenders() {
+	eventSenders = make([]EventSender, 0, 1)
+}
+
+// RegisterEventSender will register a EventSender to be used.
+func RegisterEventSender(es EventSender) {
+	eventSenders = append(eventSenders, es)
+}
+
 // SendEvent will send cloud event to given url
-func SendEvent(e cloudevents.Event, url string) error {
+func SendEvent(ce cloudevents.Event, endpoint interface{}) error {
+	senders := make([]EventSender, 0, len(eventSenders)+1)
```

Comment:
Why not just `var senders []EventSender` ?

Reply:
Same answer as https://github.com/knative/eventing/pull/4815#discussion_r635174556